### PR TITLE
CDS annotation parser package moved from UX Tools

### PR DESCRIPTION
**Pull request by vadson71**

### CODE REVIEW

File: packages/cds-annotation-parser/README.md [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+
+Yarn
+`yarn add @sap-ux/cds-annotation-parser`
+
+Pnpm
+`pnpm add @sap-ux/cds-annotation-parser`
+
+## Usage
+
+Cds annotation file parsing. Abstract syntax tree nodes search examples.
```
Comment (voicis) [inline packages/cds-annotation-parser/README.md:15]: > Abstract syntax tree nodes search examples.
Something seems to be missing here.

File: packages/cds-annotation-parser/README.md [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+    if (propertyValueNode.type === 'path') {
+        // expected value: "some.path"
+        const value = propertyValueNode.value;
+    }
+}
+
+```
+
+## Keywords
+OData annotations
```
Comment (voicis) [inline packages/cds-annotation-parser/README.md:75]: `CAP CDS` could also be added here.
```suggestion
OData annotations CAP CDS
```

File: packages/cds-annotation-parser/package.json [hunk 3]
```diff
@@ -0,0 +1,64 @@
+{
+    "name": "@sap-ux/cds-annotation-parser",
+    "version": "0.1.0",
+    "description": "SAP Fiori tools - Annotation Parser for CDS",
```
Comment (voicis) [inline packages/cds-annotation-parser/package.json:4]:
```suggestion
    "description": "Annotation Parser for CDS",
```

File: packages/cds-annotation-parser/package.json [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+        "clean:dist": "rimraf ./dist ./generators *.tsbuildinfo",
+        "clean": "rimraf ./reports",
+        "format:fix": "prettier --write --loglevel silent --ignore-path ../../../.prettierignore",
+        "format:fix:all": "prettier --write '**/*.{css,scss,html,js,json,ts,tsx,yaml,yml}' '!**/{out,dist,typings,node_modules}/**' '!**/*.{svg,png,xml}' --ignore-path ../../../.prettierignore",
+        "lint": "eslint . --ext .ts,.tsx",
+        "lint:fix": "eslint . --ext .ts --fix",
+        "lint:fix:all": "eslint . --ext .ts,.tsx --fix",
+        "lint:report": "eslint . --ext .ts,.tsx -f multiple ",
+        "lint:summary": "eslint . --ext .ts,.tsx -f summary",
+        "madge": "madge --warning --circular --extensions ts ./",
```
Comment (voicis) [inline packages/cds-annotation-parser/package.json:21]: This is not used here.

File: packages/cds-annotation-parser/package.json [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+    "version": "0.1.0",
+    "description": "SAP Fiori tools - Annotation Parser for CDS",
+    "publisher": "SAPSE",
+    "author": "SAP SE",
+    "license": "Apache-2.0",
+    "main": "dist/index.js",
+    "scripts": {
+        "build": "npm-run-all clean compile",
+        "compile": "tsc --build ./tsconfig.build.json",
+        "clean:dist": "rimraf ./dist ./generators *.tsbuildinfo",
```
Comment (voicis) [inline packages/cds-annotation-parser/package.json:12]: There is no generators folder here.
```suggestion
        "clean:dist": "rimraf ./dist *.tsbuildinfo",
```

File: packages/cds-annotation-parser/src/transformer/annotationAstNodes.ts [hunk 6]
```diff
@@ -0,0 +1,279 @@
+import { Range } from 'vscode-languageserver-types';
```
Comment (voicis) [inline packages/cds-annotation-parser/src/transformer/annotationAstNodes.ts:1]: Change file name to use kebab case.

File: packages/cds-annotation-parser/src/transformer/annotationAstNodes.ts [hunk 7]
```diff
@@ -0,0 +1,279 @@
...
+    type: typeof OPERATOR_TYPE;
+    value: string; // image of token representing operator (e.g. '-')
+    range: Range;
+}
+
+export type EXPRESSION_TYPE =
+    | typeof INCORRECT_EXPRESSION_TYPE
+    | typeof UNSUPPORTED_OPERATOR_EXPRESSION_TYPE
+    | typeof CORRECT_EXPRESSION_TYPE;
+export interface Expression extends Node {
```
Comment (voicis) [inline packages/cds-annotation-parser/src/transformer/annotationAstNodes.ts:103]: Please add the fix that was needed for CDS converter here already.
We should have `BaseExpression` interface without type and `Expression` type should be a union type with the 3 kinds of expressions.

File: packages/cds-annotation-parser/tsconfig.json [hunk 8]
```diff
@@ -0,0 +1,33 @@
+{
+  "extends": "../../tsconfig.json",
+  "compilerOptions": {
+    "outDir": "./dist",
+    "rootDir": "./src",
+    "lib": [
+      "ES2021"
+    ],
+    "forceConsistentCasingInFileNames": true,
+    "noImplicitAny": false,
```
Comment (voicis) [inline packages/cds-annotation-parser/tsconfig.json:10]: This option should be `true` and `strict` mode should also be enabled.

File: packages/cds-annotation-parser/test/data/invalid/expression/orphaned operand/ast.json [hunk 9]
```diff
@@ -0,0 +1,89 @@
+{
```
Comment (voicis) [inline packages/cds-annotation-parser/test/data/invalid/expression/orphaned operand/ast.json:1]: We shouldn't have whitespace in folder names.

File: .gitattributes [hunk 10]
```diff
@@ -0,0 +1,34 @@
+# Some standard definitions
+* text eol=lf
+*.cmd eol=crlf
+*.sh eol=lf
+*.bat eol=crlf
+*.ps1 eol=crlf
+*.json-dos eol=crlf
+*.cjs eol=lf
+yarn.lock eol=lf merge=ours
```
[the remark this answers is not in the document]
Comment (vadson71) [reply .gitattributes:9]: Changed to pnpm-lock.yaml

File: packages/cds-annotation-parser/README.md (hunk 2 above)
Comment (vadson71) [reply packages/cds-annotation-parser/README.md:75]: Added

File: packages/cds-annotation-parser/package.json (hunk 3 above)
Comment (vadson71) [reply packages/cds-annotation-parser/package.json:4]: Done

File: packages/cds-annotation-parser/package.json (hunk 4 above)
Comment (vadson71) [reply packages/cds-annotation-parser/package.json:21]: Deleted

File: packages/cds-annotation-parser/package.json (hunk 5 above)
Comment (vadson71) [reply packages/cds-annotation-parser/package.json:12]: Removed

File: packages/cds-annotation-parser/src/transformer/annotationAstNodes.ts (hunk 6 above)
Comment (vadson71) [reply packages/cds-annotation-parser/src/transformer/annotationAstNodes.ts:1]: done

File: packages/cds-annotation-parser/src/transformer/annotationAstNodes.ts (hunk 7 above)
Comment (vadson71) [reply packages/cds-annotation-parser/src/transformer/annotationAstNodes.ts:103]: Done

File: packages/cds-annotation-parser/test/data/invalid/expression/orphaned operand/ast.json (hunk 9 above)
Comment (vadson71) [reply packages/cds-annotation-parser/test/data/invalid/expression/orphaned operand/ast.json:1]: renamed

File: packages/cds-annotation-parser/README.md (hunk 1 above)
Comment (vadson71) [reply packages/cds-annotation-parser/README.md]: Updated

File: packages/cds-annotation-parser/CONTRIBUTING.MD [hunk 11]
```diff
@@ -0,0 +1,25 @@
+### CDS Annotation Parser Tests
+
+Test data can be found in `test/data` folder with sub-folders `valid` and `invalid`. Each test case has its respective folder or sub-folder(s) which contains:
+1. `assignment.txt` with annotation assignment source text
+2. `cst.json` compact form of CST (Context Syntax Tree) for given source text
+3. `ast.json` compact form of AST (Abstract Syntax Tree) for the given source text
+
+### Writing a new test
+1. open a terminal and navigate to `packages/lib/cds-annotation-parser`
```
Comment (devinea) [inline packages/cds-annotation-parser/CONTRIBUTING.MD:9]:
```suggestion
1. open a terminal and navigate to `packages/cds-annotation-parser`
```
correct paths

File: packages/cds-annotation-parser/package.json [hunk 12]
```diff
@@ -0,0 +1,63 @@
...
+        "clean:dist": "rimraf ./dist *.tsbuildinfo",
+        "clean": "rimraf ./reports",
+        "format:fix": "prettier --write --loglevel silent --ignore-path ../../../.prettierignore",
+        "format:fix:all": "prettier --write '**/*.{css,scss,html,js,json,ts,tsx,yaml,yml}' '!**/{out,dist,typings,node_modules}/**' '!**/*.{svg,png,xml}' --ignore-path ../../../.prettierignore",
+        "lint": "eslint . --ext .ts,.tsx",
+        "lint:fix": "eslint . --ext .ts --fix",
+        "lint:fix:all": "eslint . --ext .ts,.tsx --fix",
+        "lint:report": "eslint . --ext .ts,.tsx -f multiple ",
+        "lint:summary": "eslint . --ext .ts,.tsx -f summary",
+        "pre-commit": "lint-staged --quiet",
```
Comment (devinea) [inline packages/cds-annotation-parser/package.json:21]:
```suggestion
```
I think these are not applicable here?

File: packages/cds-annotation-parser/package.json [hunk 13]
```diff
@@ -0,0 +1,63 @@
...
+                "output": "file",
+                "path": "reports/lint/eslint.json"
+            },
+            {
+                "name": "checkstyle",
+                "output": "file",
+                "path": "reports/lint/eslint.checkstyle.xml"
+            }
+        ]
+    }
```
Comment (devinea) [inline packages/cds-annotation-parser/package.json:62]:
```suggestion
```
Not needed here and can also be removed from the other annotation modules.

**Review by devinea — COMMENTED** [review]
Set-up for new module generally looks good. 
A few small comments to be addressed.

File: packages/cds-annotation-parser/CONTRIBUTING.MD (hunk 11 above)
Comment (vadson71) [reply packages/cds-annotation-parser/CONTRIBUTING.MD]: fixed

File: packages/cds-annotation-parser/package.json (hunk 12 above)
Comment (vadson71) [reply packages/cds-annotation-parser/package.json]: Yes, true. Deleted

File: packages/cds-annotation-parser/package.json (hunk 13 above)
Comment (vadson71) [reply packages/cds-annotation-parser/package.json]: Removed

**Review by vinayhospete — APPROVED** [review]

**Review by nahmed22 — APPROVED** [review]
- Changeset maintained
- Good test coverage

**Review by devinea — APPROVED** [review]
Thanks for addressing the comments.
New
Module setup looks
good from my side.
I haven't reviewed the content and functionality of the module so much.

**Pull request merged**